### PR TITLE
Rules - GHE Required Status Checks, CloudTrail Bucket ACL fix, and Cleanup

### PR DESCRIPTION
**Pull request by mime-frame**
to: @jacknagz 
cc: @airbnb/streamalert-maintainers
size: medium

## Background

* New Github rule for disabling required status checks for a repo
* Syncing internal changes
* Cleanup

## Changes

* New Github rule for disabling required status checks for a repo
* Fixed S3 Bucket ACL rule with associated test (sync)
* Update the GHE schemas to the new ones we have internally
* Remove the old matcher and uses of it (like we did internally)
* Update the rules to no longer use the GHE helper function
* Remove the GHE helper function
* Create integration test folders for onelogin and mitre_attack
* Update types.json to account for GHE fields like `actor_ip`

## Testing

```
StreamAlertCLI [INFO]: (42/42) Successful Tests
StreamAlertCLI [INFO]: (72/72) Alert Tests Passed
StreamAlertCLI [INFO]: Completed
```

### CODE REVIEW

**Comment by mime-frame** [comment]
Alrighty, this is now ready for review @jacknagz

**Comment by mime-frame** [comment]
@jacknagz I'm only syncing the bucket ACL rule in this PR and will do the other cloudtrail rules in another PR, as they're more cumbersome and require some surgery (ex: the consts that exist outside the rule)

File: rules/community/github/github_disable_required_status_checks.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+def github_disable_required_status_checks(rec):
+    """
+    author:       @mimeframe
+    description:  The 'required status checks' feature was disabled for a repository.
+                  Settings -> Branches -> Protected Branches -> <choose a branch>
+    reference:    https://help.github.com/articles/enabling-required-status-checks/
+    """
+    return (
+        rec['action'] == 'protected_branch.update_required_status_checks_enforcement_level' and
+        rec['data'].get('required_status_checks_enforcement_level') == '0'
```
Comment (jacknagz) [inline rules/community/github/github_disable_required_status_checks.py]: are you sure this is a string and not an int?

Comment (mime-frame) [reply rules/community/github/github_disable_required_status_checks.py]: Good catch. Snippet from log: `"_document_id":"...","required_status_checks_enforcement_level":1,"actor_location":{...`

Address this now

**Pull request merged**